dbt/adapters/athena/impl.py

```diff
@@ -203,55 +216,54 @@ def s3_table_location(
+        schema_name = relation.schema
+        s3_data_naming = self._s3_data_naming(s3_data_naming)
+        table_prefix = self._s3_table_prefix(s3_data_dir)
 
         mapping = {
-            "uuid": path.join(self.s3_table_prefix(s3_data_dir), str(uuid4())),
-            "table": path.join(self.s3_table_prefix(s3_data_dir), s3_path_table_part),
-            "table_unique": path.join(self.s3_table_prefix(s3_data_dir), s3_path_table_part, str(uuid4())),
-            "schema_table": path.join(self.s3_table_prefix(s3_data_dir), schema_name, s3_path_table_part),
-            "schema_table_unique": path.join(
-                self.s3_table_prefix(s3_data_dir), schema_name, s3_path_table_part, str(uuid4())
-            ),
+            S3DataNaming.UUID: path.join(table_prefix, str(uuid4())),
+            S3DataNaming.TABLE: path.join(table_prefix, s3_path_table_part),
+            S3DataNaming.TABLE_UNIQUE: path.join(table_prefix, s3_path_table_part, str(uuid4())),
+            S3DataNaming.SCHEMA_TABLE: path.join(table_prefix, schema_name, s3_path_table_part),
+            S3DataNaming.SCHEMA_TABLE_UNIQUE: path.join(table_prefix, schema_name, s3_path_table_part, str(uuid4())),
         }
-        table_location = mapping.get(s3_data_naming)
 
-        if table_location is None:
-            raise ValueError(f"Unknown value for s3_data_naming: {s3_data_naming}")
-
-        return table_location
+        return mapping[s3_data_naming]
 
     @available
-    def get_table_location(self, database_name: str, table_name: str) -> Union[str, None]:
+    def get_glue_table_location(self, relation: AthenaRelation) -> Optional[str]:
         """
-        Helper function to S3 get table location
+        Helper function to get location of a relation in S3.
+        Will return nothing if the table does not exist
+        Will return empty if the relation is a view
         """
         conn = self.connections.get_thread_connection()
         client = conn.handle
         with boto3_client_lock:
             glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
         try:
-            table = glue_client.get_table(DatabaseName=database_name, Name=table_name)
-            table_location = table["Table"]["StorageDescriptor"]["Location"]
-            logger.debug(f"{database_name}.{table_name} is stored in {table_location}")
-            return table_location
+            table = glue_client.get_table(DatabaseName=relation.schema, Name=relation.identifier)
         except ClientError as e:
             if e.response["Error"]["Code"] == "EntityNotFoundException":
-                logger.debug(f"Table '{table_name}' does not exists - Ignoring")
+                logger.debug(f"Table {relation.render()} does not exists - Ignoring")
                 return
+            raise e
+        table_location = table["Table"]["StorageDescriptor"]["Location"]
```

Comment:
I like to keep the body of a try/except as slim as possible. Since the exception is only for the glue call, I want only the glue call in the try block.

For a view the function will not fail but will return an empty string, therefore the pruning won't happen either.

Reply:
>I like to keep the body of a try/except as slim as possible. Since the exception is only for the glue call, I want only the glue call in the try block.
me too

> For a view the function will not fail but will return an empty string, therefore the pruning won't happen either. And it is the same behavior as before. Glue does not raise exception when gettting table on a view.
is location then None? in case of view?

Reply:
It is an empty string.

And to circle back to the first comment:
```
 table_location = table["Table"]["StorageDescriptor"]["Location"]
 ```
 
If there was no location in the StorageDescription, this was not catch by the exception `ClientError` because once we have our response, the only exception I can see occurring is the `KeyError` no ?

Therefore I think it makes the code more comprehensive to move this out of the exception block.

Reply:
What we can do is, using the `get_table_type` check :
- If this is a table, location must be set
- If this is a view, we can choose the return empty string or None

In any way, this function is only used to know if we need to clean up s3 location and we check

```
if table_location
```

Therefore empty string and None are treated identically.

Reply:
>Therefore empty string and None are treated identically. 
yes, but it will be easy if we catch emptry strings (aka views) and return None. Doing so we have cleaner check downstreams wdyt?

```
if table_location == '':
  return None
```

Reply:
Totally agree 👍🏻

Reply:
As you can see, I used the `get_table_type` to make an enhanced check about the location.